README.rst

```diff
@@ -26,39 +26,53 @@ This is easily achieved by downloading
 Usage Example
 =============
 
-The current `keyboard` module stores key constants which make it easier to
-construct keypress reports for a keyboard device.
+The ``keyboard`` module defines keycode constants and a ``Keyboard`` class to
+construct and send keypress reports for a USB keyboard device.
 
 .. code-block:: python
 
-    import usb_hid
-    import adafruit_hid.keyboard as kbd
-    import time
-
-    report = bytearray(8) # Keyboard reports are always 8 bytes.
-
-    # Devices are initialized earlier so find the one for the keyboard.
-    keyboard = None
-    for device in usb_hid.devices:
-        if device.usage_page is 0x1 and device.usage is 0x06:
-            keyboard = device
-            break
-
-    # The first byte of the report includes a bitfield indicating which
-    # modifiers are pressed. Their bit position is their code's difference from
-    # 0xE0.
-    report[0] |= 1 << (kbd.LEFT_SHIFT - 0xE0)
-    # Normal keys are simply their byte code in bytes 2-7. When fewer than six
-    # keys are pressed then the trailing bytes are zero.
-    report[2] = kbd.A
-    keyboard.send_report(report)
-
-    time.sleep(0.1)
-
-    # Clear the key presses and send another report.
-    report[0] = 0
-    report[2] = 0
-    keyboard.send_report(report)
+    from adafruit_hid.keyboard import Keyboard
+
+    # Find a keyboard device to talk to.
+    kbd = Keyboard()
+
+    # Simulate typing. Press and release the A key (not shifted),
+    # then the B key, then c, then the Enter key.
+    # Type "abc" followed by return.
+    kbd.type("abc\n")
+
+    # Type control-x, then "Abc", then backspace.
+    kbd.type((Keyboard.CONTROL, 'x'), 'Abc', Keyboard.BACKSPACE)
```

Comment:
I see the package providing three pieces of functionality:

1. Press and release keycodes.
2. Mapping from character codes to keycodes.
3. Convenience methods for ordinary keyboard typing.

If the user is doing #1 but not #3, s/he will still need to do #2 if the program is doesn't know until run-time what it is going to type. I like your idea of possible future alternate layouts. The Arduino library is very US-keyboard-centric.

Originally I wrote the `_char_to_keycode()` method to return a two-element tuple of `(modifier, keycode)`. I took that out because it allocated storage and I wanted to avoid that. The `ASCII_TO_KEYCODE` table is only 128 bytes because of the bit-encoding for shift, and it can live in flash, but it only works for English.

But the general case is a table of tuples, or similar. A Unicode character may require multiple keypresses and releases with modifiers. For instance, on a Mac, to type `è`, you can type `Option` + `` ` `` then `e`. So might be represented as `((OPTION, GRAVE_ACCENT), E)` or even `((OPTION, GRAVE_ACCENT), (E,))`where each top-level element is a combination to press and release. Or if you used dead keys (as on Windows, I think), it could be `(GRAVE_ACCENT, E)`.

So maybe I should give up on the idea of not generating garbage and using RAM for the lookup tables. There's an efficiency/generality tradeoff here that's difficult given the tight storage constraints.